src/storage/animationsSchema.ts

```diff
@@ -0,0 +1,449 @@
+		atLocation: z
+			.object({
+				cacheLocation: z.literal(true).optional(),
+				offset: vector2.optional(),
+				randomOffset: z.number().optional(),
+				gridUnits: z.literal(true).optional(),
+				local: z.literal(true).optional(),
+			})
+			.strict()
+			.refine(...nonEmpty)
+			.optional(),
+		stretchTo: z
+			.object({
+				cacheLocation: z.literal(true).optional(),
+				attachTo: z.literal(true).optional(),
+				onlyX: z.literal(true).optional(),
+				tiling: z.literal(true).optional(),
+				offset: vector2.optional(),
+				randomOffset: z.number().optional(),
+				gridUnits: z.literal(true).optional(),
+				local: z.literal(true).optional(),
+				requiresLineOfSight: z.literal(true).optional(),
+				hideLineOfSight: z.literal(true).optional(),
+			})
+			.strict()
+			.refine(...nonEmpty)
+			.optional(),
+		rotateTowards: z
+			.object({
+				rotationOffset: z.number().optional(),
+				cacheLocation: z.literal(true).optional(),
+				attachTo: z.literal(true).optional(),
+				offset: vector2.optional(),
+				randomOffset: z.number().optional(),
+				gridUnits: z.literal(true).optional(),
+				local: z.literal(true).optional(),
+			})
+			.strict()
+			.refine(...nonEmpty)
+			.optional(),
+		anchor: vector2.optional(),
+		template: z
+			.object({
+				gridSize: z.number().min(0),
+				startPoint: z.number(),
+				endPoint: z.number(),
+			})
+			.strict()
+			.optional(),
+		loopProperty: z
+			.array(
+				z
+					.object({
+						target: z.string(),
+						property: z.string(),
+						options: z
+							.object({
+								duration: z.number(),
+								from: z.number().optional(),
+								to: z.number().optional(),
+								values: z.array(z.number()).min(1).optional(),
+								loops: z.number().optional(),
+								pingPong: z.literal(true).optional(),
+								delay: z.number().optional(),
+								ease: z.string().optional(),
+								fromEnd: z.literal(true).optional(),
+								gridUnits: z.literal(true).optional(),
+							})
+							.strict(),
+					})
+					.strict(),
+			)
+			.min(1)
+			.optional(),
+		animateProperty: z
+			.array(
+				z
+					.object({
+						target: z.string(),
+						property: z.string(),
+						options: z
+							.object({
+								duration: z.number(),
+								from: z.number(),
+								to: z.number(),
+								delay: z.number().optional(),
+								ease: z.string().optional(),
+								fromEnd: z.literal(true).optional(),
+								gridUnits: z.literal(true).optional(),
+							})
+							.strict(),
+					})
+					.strict(),
+			)
+			.min(1)
+			.optional(),
+		shape: shape.or(z.array(shape).min(1)).optional(),
+	})
+	.strict()
+	.refine(...nonEmpty)
+
+const animationDataObject = z.object({
+	overrides: z.array(rollOption).min(1).optional(),
+	trigger: z.enum(AnimCore.CONST.TRIGGERS),
+	preset: z.enum(Object.keys(presets) as [string, ...string[]]),
+	file: fileName,
+	default: z.literal(true).optional(),
+	predicate: z.array(predicate).min(1).optional(),
+	options: z.any().optional(), // :(
```

Comment:
The `any` here (and lack of `strict()` are due to a lack of strict definition in `AnimCore.ts#421`.

Reply:
Laziness on my part and the fact that typing all that is a pain.

Given this is for validating data, you are welcome to actually check that part as its just all the options combined.